code/drasil-code/lib/Language/Drasil/Choices.hs

```diff
@@ -362,4 +366,27 @@ choicesSent chs = map chsFieldSent [
 
 -- | Helper to combine pairs of 'Sentence's for rendering 'Choices'.
 chsFieldSent :: (Sentence, Sentence) -> Sentence
-chsFieldSent (rec, chc) = rec +:+ S "selected as" +:+. chc
\ No newline at end of file
+chsFieldSent (rec, chc) = rec +:+ S "selected as" +:+. chc
+
+-- | List of user defined function names.
+-- | List is populated with default values.
+fnList :: [(InternalConcept, Name)]
```

Comment:
So as long as `InternalConcept` has an Eq instance, I think we can also make this a `Map InternalConcept Name` to avoid conflicting entries. Alternatively, it could be a provided Haskell-level function.

Reply:
@balacij Are you suggesting a helper function for editing the entries in `fnList`? How about the following:

```
-- | Helper for list entries
fnListEntry :: Eq k => k -> v -> [(k, v)] -> [(k, v)]
fnListEntry key value list = (key,value):(filter ((key /=).fst) list)
```

Reply:
That does look like it would do what I suggested, but I was suggesting we can use [Data.Map's Map](https://hackage.haskell.org/package/containers-0.6.5.1/docs/Data-Map-Lazy.html) (Lazy/Strict, probably doesn't matter here since it's small) to contain the mapping, and then we can use their tools a bit. Then, the fnList type signature would change to `Map InternalConcept Name` (e.g., a mapping from InternalConcepts to Name). The `Map k v` requires the type `k` to have an `Eq` instance defined (which, I now see you have defined through the `deriving Eq` bit in the data definition). Alternatively, a haskell-level function might be okay too (e.g., `fnList :: InternalConcept -> Name`).

Reply:
Yes, `Data.Map` would be better here. Better reuse.